manifests/agent/service/systemd.pp

```diff
@@ -0,0 +1,67 @@
+class puppet::agent::service::systemd (
+  $enabled = false,
+) {
+
```

Comment:
Yeah, I think the `unless` clauses should be inside the classes - they're just implementation details. That way they're properly skipped by tests that can't run them.